backend/shared/src/main/scala/zio/logging/LogFormat.scala

```diff
@@ -0,0 +1,214 @@
+    highlight(defaultHighlighter(_))
+
+  final def highlight(fn: LogLevel => LogColor)(implicit ev: Out <:< String): LogFormat[String] = {
+    val self = this.map(ev)
+    LogFormat { (builder, trace, fiberId, level, line, fiberRefs, spans) =>
+      builder(fn(level).ansi)
+      self.unsafeFormat(builder)(trace, fiberId, level, line, fiberRefs, spans)
+      builder(LogColor.RESET.ansi)
+    }
+  }
+
+  final def map[Out2](fn: Out => Out2): LogFormat[Out2] = new LogFormat[Out2] {
+    override private[logging] def unsafeFormat(builder: Out2 => Unit) =
+      self.unsafeFormat((out: Out) => builder(fn(out)))
+
+    override def toLogger: ZLogger[Out2] = self.toLogger.map(fn)
+  }
+
+  final def spaced(other: LogFormat[String])(implicit ev: Out <:< String): LogFormat[String] =
+    this |-| other
+
+  def toLogger: ZLogger[Out]
+}
+
+object LogFormat {
+
+  private val NL = System.lineSeparator()
+
+  private[logging] def apply(
+    format: (
+      String => Unit,
+      ZTraceElement,
+      FiberId,
+      LogLevel,
+      () => String,
+      Map[ZFiberRef.Runtime[_], AnyRef],
+      List[LogSpan]
+    ) => Any
+  ): LogFormat[String] = new LogFormat[String] {
+    override private[logging] def unsafeFormat(builder: String => Unit) = new ZLogger[Unit] {
+      override def apply(
+        trace: ZTraceElement,
+        fiberId: FiberId,
+        logLevel: LogLevel,
+        message: () => String,
+        context: Map[ZFiberRef.Runtime[_], AnyRef],
+        spans: List[LogSpan]
+      ): Unit = {
+        format(builder, trace, fiberId, logLevel, message, context, spans)
+        ()
+      }
+    }
+
+    override def toLogger: ZLogger[String] = new ZLogger[String] {
+      override def apply(
+        trace: ZTraceElement,
+        fiberId: FiberId,
+        logLevel: LogLevel,
+        message: () => String,
+        context: Map[ZFiberRef.Runtime[_], AnyRef],
+        spans: List[LogSpan]
+      ): String = {
+        val builder = new StringBuilder()
+        //FIXME this should not be required when core will handle exception properly.
+        try unsafeFormat(builder.append(_: String))(trace, fiberId, logLevel, message, context, spans)
+        catch {
+          case ex: Throwable => ex.printStackTrace()
+        }
+
+        builder.toString()
+      }
+    }
+  }
+
+  val bracketStart: LogFormat[String] = string("[")
+
+  val bracketEnd: LogFormat[String] = string("]")
+
+  val quote: LogFormat[String] = string("\"")
+
+  def quoted(inner: LogFormat[String]): LogFormat[String] = quote + inner + quote
+
+  val newLine: LogFormat[String] = string(NL)
+
+  def bracketed(inner: LogFormat[String]): LogFormat[String] =
+    bracketStart + inner + bracketEnd
+
+  val level: LogFormat[String] =
+    LogFormat { (builder, _, _, level, _, _, _) =>
+      builder(level.label)
+    }
+
+  val fiberNumber: LogFormat[String] =
+    LogFormat { (builder, _, fiberId, _, _, _, _) =>
+      builder("#")
```

Comment:
We are standardizing on `zio-fiber-<seq-num>` to format the id of ZIO fibers.

Reply:
ack